refactor: use main repository for upgrading (#3155)
oven-sh/bun: docs/cli/bun-upgrade.md

It automatically downloads the latest version of Bun and overwrites the currently-running version.

> This works by checking the latest version of Bun in [bun-releases-for-updater](https://github.com/oven-sh/bun/releases) and unzipping it using the system-provided `unzip` library (so that Gatekeeper works on macOS)

Reviewer: ```suggestion
This works by checking the latest release of Bun from GitHub [releases](https://github.com/oven-sh/bun/releases) and unzipping it using the system-provided `unzip` library (so that Gatekeeper works on macOS)
```
